File: src/main/kotlin/a/HelpfulMaths.kt
package a

/* Problem Link: https://codeforces.com/problemset/problem/339/A */

fun main() {
    val calculation = readln()
    val numbers = calculation.split("+").sorted()
    numbers.forEachIndexed { index, s ->
        print(s)
        if (index != numbers.size - 1) print("+")
    }
}

File: Codeforces/C++/problemset/Helpful Maths.cpp
// Problem Link: https://codeforces.com/problemset/problem/339/A

#include <iostream>
#include <string>
using namespace std;
typedef long long ll;
int main()
{
    string s, r = "";
    cin >> s;

    for(ll i = 0; i < s.size(); i++)
    {
        if(s[i] != '+') r+=s[i];
    }

    for(ll i = 0; i < r.size()-1; i++)
    {
        for(ll j = 0; j < r.size()-i-1; j++)
        {
            if(r[j] > r[j+1]) swap(r[j], r[j+1]);
        }
    }

    for(ll i = 0; i < r.size(); i++)
    {
        if(i == r.size()-1) cout << r[i];
        else cout << r[i] << '+';
    }

    return 0;
}
